central/notifier/datastore/datastore_impl.go

```diff
@@ -245,3 +261,23 @@ func (b *datastoreImpl) RemoveNotifier(ctx context.Context, id string) error {
 	}
 	return b.storage.Delete(ctx, id)
 }
+
+func validateNotifier(notifier *storage.Notifier) error {
```

Comment:
This validation performs only a subset of validations compared to the one defined under `stackrox/rox/central/notifiers/validation`. Why did you not choose to reuse the `ValidateNotifierConfig`?

Reply:
Two reasons: the additional edge-cases for setting `validateSecret` or not is currently not exposed within the datastore functions, and potentially would need to be exposed. Second is that it creates an import cycle, and since the validation is already done at the client side (i.e. service and declarative config updater) I skipped it. Note that the `ValidateNotifierConfig` does not do any of the validation within the datastore (name, type, endpoint).